oauthenticator/github.py

```diff
@@ -113,6 +114,21 @@ def _github_client_secret_changed(self, name, old, new):
         config=True, help="Automatically allow members of selected organizations"
     )
 
+    fetch_teams = Bool(
```

Comment:
`fetch_teams` is short and brief and I accept it, but I'd prefer something more verbose if you think it also could make sense @yuvipanda. For example: `add_teams_to_auth_state` or `fetch_teams_to_auth_state`.

I appreciate that `_to_auth_state` clarifies the end result, I appreciate that `fetch` implies notable work, I appreciate that `add_teams` more directly clarifies the result that `teams` will be in `auth_state` than `fetch_teams`.

Feel free to resolve without action if you are opinionated towards `fetch_teams`.

Reply:
`fetch_teams` is definitely too vague - thanks for catching that! I've renamed it to `populate_team_membership`, which I think is definitely an improvement as it tells you what it does after fetching. However, I'd like to keep `auth_state` out of it as it feels a bit too verbose to me. If you feel strongly about it I can try find a way to work it in!

Reply:
Somehow this particular name has bothered me, and I kept trying to figure out why. While I still can not articulate exactly why, it's given me an idea on how to actually redo this PR in a way that's more generic, where things are easier to name! I'll work on it, and I've converted this PR to draft in the meantime.

Reply:
@consideRatio ok i clearly couldn't actually redo this PR. Is calling it `populate_team_membership` acceptable to you? If so, this is ready for merge.